contexts/WalletManagerContext.tsx

```diff
@@ -567,36 +566,36 @@ export function WalletManagerProvider({ children }: { children: ReactNode }) {
     await keyring.keyringReset();
   };
 
-  // Refresh accounts after settings changed
-  useEffect(() => {
-    if (!rpcClient) {
-      return;
+  const refreshAccounts = async () => {
+    if (!networkId) {
+      throw new Error("RPC client and settings not loaded");
     }
 
-    const refreshAccounts = async () => {
-      if (!networkId) {
-        throw new Error("RPC client and settings not loaded");
-      }
-
-      for (const wallet of walletSettings.wallets) {
-        for (const account of wallet.accounts) {
-          // TODO: Remove this after the next release
-          await generatePublicKeysForOldVersion(wallet, account.index);
-
-          // hotfix for missing public keys
-          if (!account.publicKeys?.length) {
-            continue;
-          }
+    for (const wallet of walletSettings.wallets) {
+      for (const account of wallet.accounts) {
+        // TODO: Remove this after the next release
```

Comment:
It would be better to remove this comment since we couldn't know when user upgrade their app.